src/client/datascience/notebook/notebookControllerManager.ts

```diff
@@ -231,6 +257,9 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
         document: NotebookDocument,
         token: CancellationToken
     ): Promise<KernelConnectionMetadata | undefined> {
+        // Don't look for a preferred kernel until we have finished our initial controller load
```

Comment:
This seems out of place here. It has nothing to do with finding a preferred kernel. Perhaps the caller of this should make this call?

Reply:
@rchiodo Sorry I had just been waiting for Don's review to push, so I pushed before this comment came in. I actually agree with you, it's a precondition for setting the preferredController, but it's not a precondition for finding a preferred kernel. I'll do a follow up PR.